app/actions/local/post.ts

```diff
@@ -137,27 +137,28 @@ export const selectAttachmentMenuAction = (serverUrl: string, postId: string, ac
     return postActionWithCookie(serverUrl, postId, actionId, '', selectedOption);
 };
 
-export const markPostAsDeleted = async (serverUrl: string, post: Post) => {
+export const markPostAsDeleted = async (serverUrl: string, post: Post, prepareRecordsOnly = false) => {
```

Comment:
I wonder if we should take care of the "delete post" action (not sure exactly were).

My point is:
- When I create a post, I create a temporal post which sooner or later will be marked as posted or failed.
  - Does the thread gets updated with that post?
- If the posts fail, I mark it as failed.
  - Should update the thread as if I have one less post?
- When we click on the failed post, we can resend it, or delete it.
  - Is all that properly handled?
- When a post is deleted by websocket, we can click it to completely delete it from the database.
  - We shouldn't do anything on threads in this case, right? (It should be already handled by the websocket delete)

Reply:
We create a thread record for every rootPost only after it gets the `id` from  the server. In case it fails to post, we don't do anything.

Yes, when the post record is deleted, it handles deleting the thread records (not when marking it as delete).

We are preparing `markPostAsDeleted` only to batch the `thread` `replyCount` operator.